modules/modal/react/lib/ModalContent.tsx

```diff
@@ -178,17 +209,21 @@ const ModalContent = ({
       ref={modalRef}
       onClick={mergeCallback(onOverlayClick, elemProps.onClick)}
     >
-      <Popup
-        width={width}
-        heading={heading}
-        handleClose={handleClose}
-        padding={padding}
-        transformOrigin={transformOrigin}
-        aria-modal={true}
-        ariaLabel={ariaLabel}
+      <CenteringContainer
+        style={{width: windowSize.width % 2 === 1 ? 'calc(100vw - 1px)' : '100vw'}}
```

Comment:
One of the worse things about the `styled` API is what it does for dynamic styles. Emotion will inject or remove stylesheets from the page when a dynamic style changes which will trigger a full-page style recalculation. Sometimes that doesn't matter because the styles don't change often, but the habit can have huge negative performance implications. An inline style for critical styles prevents full page style recalculations.

Reply:
@NicholasBoll I feel like this comment should be inline, otherwise it could easily be mistakenly turned into `css` or a styled component prop

Reply:
I think that should be documented at a higher level. I helped another team with a major performance issue when using `styled`/`css` for x/y coordinates while dragging. The default is to put dynamic styles in Emotion, but we should probably recommends putting dynamic styles in a `style` attribute instead. I think certain properties are dynamic for the sake of the API, but won't change in practice. Like theme or our design system constants.

Maybe a best practice around using Emotion or CSS-in-JS. I could add that comment here, but it would likely be missed and we accidentally introduce performance problems in applications where we may not notice as much because our examples have less styles and DOM elements than an application.